### PR TITLE
feat(storagetransfer): Add S3 compatible transfer sample

### PR DESCRIPTION
**Pull request by JesseLovelace**
Add S3 compatible transfer sample, for this page:
https://cloud.google.com/storage-transfer/docs/s3-compatible#rest-api_1

### CODE REVIEW

File: storagetransfer/go.sum [hunk 1]
```diff
@@ -31,6 +31,12 @@ cloud.google.com/go v0.99.0/go.mod h1:w0Xx2nLzqWJPuozYQX+hFfCSI8WioryfRDzkoI/Y2Z
 cloud.google.com/go v0.100.1/go.mod h1:fs4QogzfH5n2pBXBP9vRiU+eCny7lD2vmFZy79Iuw1U=
 cloud.google.com/go v0.100.2 h1:t9Iw5QH5v4XtlEQaCtUY7x6sCABps8sW0acw7e2WQ6Y=
 cloud.google.com/go v0.100.2/go.mod h1:4Xra9TjzAeYHrl5+oeLlzbM2k3mjVhZh4UqTZ//w99A=
+cloud.google.com/go v0.102.0/go.mod h1:oWcCzKlqJ5zgHQt9YsaeTY9KzIvjyy0ArmiBUgpQ+nc=
```
Comment (tritone) [inline storagetransfer/go.sum:34]: Run `go mod tidy` in storagetransfer/ to fix the lint.

File: storagetransfer/transfer_from_s3_compatible_source.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storagetransfer
+
+import (
+	storagetransfer "cloud.google.com/go/storagetransfer/apiv1"
```
Comment (tritone) [inline storagetransfer/transfer_from_s3_compatible_source.go:18]: These imports should be sorted with the builtin packages first.

File: storagetransfer/transfer_from_s3_compatible_source.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+	"context"
+	"fmt"
+	storagetransferpb "google.golang.org/genproto/googleapis/storagetransfer/v1"
+	"io"
+)
+
+// [START storagetransfer_transfer_from_s3_compatible_source]
```
Comment (tritone) [inline storagetransfer/transfer_from_s3_compatible_source.go:25]: Region tag should come before imports.

File: storagetransfer/transfer_from_s3_compatible_source.go [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+func transferFromS3CompatibleSource(w io.Writer, projectID string, sourceAgentPoolName string, sourceBucketName string, sourcePath string, gcsSinkBucket string, gcsPath string) (*storagetransferpb.TransferJob, error) {
+	// Your project id
```
Comment (tritone) [inline storagetransfer/transfer_from_s3_compatible_source.go:28]: Nit: all the full sentence comments should end with a period.

File: storagetransfer/transfer_from_s3_compatible_source.go [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+	// The GCS path (object prefix) to transfer data to
+	//gcsPath = "path/to/data"
+
+	// The S3 region of the source bucket
+	region := "us-east-1"
+
+	// The S3 compatible endpoint
+	endpoint := "us-east-1.example.com"
+
+	// The S3 compatible network protocol
```
Comment (tritone) [inline storagetransfer/transfer_from_s3_compatible_source.go:52]: These options are a bit inscrutable but I assume the surrounding docs will clarify which to choose?

**Review by tritone — COMMENTED** [review]
Couple small things, overall looks fine.

File: storagetransfer/transfer_from_s3_compatible_source.go (hunk 5 above)
Comment (JesseLovelace) [reply storagetransfer/transfer_from_s3_compatible_source.go:52]: The target audience here (people transferring from S3 compatible sources) should know what they mean, and yeah they're in the docs

**Review by tritone — APPROVED** [review]

**Pull request merged**